### PR TITLE
regression 4003: add test cases for SM4

### PR DESCRIPTION
**Pull request by jforissier**
Adds test cases for the Chinese SM4 cipher in ECB, CBC and CTR modes.

Signed-off-by: Jerome Forissier <jerome@forissier.org>

### CODE REVIEW

**Comment by jforissier** [comment]
Depends on https://github.com/OP-TEE/optee_os/pull/3469.

File: host/xtest/regression_4000.c [hunk 1]
```diff
@@ -2143,6 +2143,155 @@ static const uint8_t ciph_data_des2_cbc_nopad_out1[] = {
...
+};
+
+static const uint8_t ciph_data_sm4_ecb_nopad_out1[] = {
+	0x68, 0x1e, 0xdf, 0x34, 0xd2, 0x06, 0x96, 0x5e,
+	0x86, 0xb3, 0xe9, 0x4f, 0x53, 0x6e, 0x42, 0x46,
+};
+
+/*
+ * SM4 CBC
+ * https://tools.ietf.org/html/draft-ribose-cfrg-sm4-10#section-8.7
```
Comment (jenswi-linaro) [inline host/xtest/regression_4000.c]: https://tools.ietf.org/html/draft-ribose-cfrg-sm4-10#appendix-A.2.2.1

File: host/xtest/regression_4000.c [hunk 2]
```diff
@@ -2143,6 +2143,155 @@ static const uint8_t ciph_data_des2_cbc_nopad_out1[] = {
...
+static const uint8_t ciph_data_sm4_cbc_a221_out[] = {
+	0x78, 0xeb, 0xb1, 0x1c, 0xc4, 0x0b, 0x0a, 0x48,
+	0x31, 0x2a, 0xae, 0xb2, 0x04, 0x02, 0x44, 0xcb,
+	0x4c, 0xb7, 0x01, 0x69, 0x51, 0x90, 0x92, 0x26,
+	0x97, 0x9b, 0x0d, 0x15, 0xdc, 0x6a, 0x8f, 0x6d,
+};
+
+/*
+ * SM4 CBC
+ * https://tools.ietf.org/html/draft-ribose-cfrg-sm4-10#section-8.7
```
Comment (jenswi-linaro) [inline host/xtest/regression_4000.c]: https://tools.ietf.org/html/draft-ribose-cfrg-sm4-10#appendix-A.2.2.2

File: host/xtest/regression_4000.c [hunk 3]
```diff
@@ -2143,6 +2143,155 @@ static const uint8_t ciph_data_des2_cbc_nopad_out1[] = {
...
+static const uint8_t ciph_data_sm4_cbc_a222_out[] = {
+	0x0d, 0x3a, 0x6d, 0xdc, 0x2d, 0x21, 0xc6, 0x98,
+	0x85, 0x72, 0x15, 0x58, 0x7b, 0x7b, 0xb5, 0x9a,
+	0x91, 0xf2, 0xc1, 0x47, 0x91, 0x1a, 0x41, 0x44,
+	0x66, 0x5e, 0x1f, 0xa1, 0xd4, 0x0b, 0xae, 0x38,
+};
+
+/*
+ * SM4 CTR
+ * https://tools.ietf.org/html/draft-ribose-cfrg-sm4-10#section-8.7
```
Comment (jenswi-linaro) [inline host/xtest/regression_4000.c]: https://tools.ietf.org/html/draft-ribose-cfrg-sm4-10#appendix-A.2.5.1

File: host/xtest/regression_4000.c [hunk 4]
```diff
@@ -2143,6 +2143,155 @@ static const uint8_t ciph_data_des2_cbc_nopad_out1[] = {
...
+	0x07, 0x4c, 0xce, 0x38, 0x5c, 0xdd, 0x70, 0xc7,
+	0xf2, 0x34, 0xbc, 0x0e, 0x24, 0xc1, 0x19, 0x80,
+	0xfd, 0x12, 0x86, 0x31, 0x0c, 0xe3, 0x7b, 0x92,
+	0x6e, 0x02, 0xfc, 0xd0, 0xfa, 0xa0, 0xba, 0xf3,
+	0x8b, 0x29, 0x33, 0x85, 0x1d, 0x82, 0x45, 0x14,
+};
+
+/*
+ * SM4 CTR
+ * https://tools.ietf.org/html/draft-ribose-cfrg-sm4-10#section-8.7
```
Comment (jenswi-linaro) [inline host/xtest/regression_4000.c]: https://tools.ietf.org/html/draft-ribose-cfrg-sm4-10#appendix-A.2.5.2

**Pull request merged**